consensus/cosipbft/mod.go

```diff
@@ -32,12 +32,17 @@ type Consensus struct {
 
 // NewCoSiPBFT returns a new instance.
 func NewCoSiPBFT(mino mino.Mino, cosi cosi.CollectiveSigning) *Consensus {
+	chainFactory := newChainFactory(cosi.GetVerifier())
+
 	c := &Consensus{
 		storage: newInMemoryStorage(),
 		mino:    mino,
 		cosi:    cosi,
-		factory: newChainFactory(cosi.GetVerifier()),
-		queue:   &queue{verifier: cosi.GetVerifier()},
+		factory: chainFactory,
+		queue: &queue{
+			verifier: cosi.GetVerifier(),
+			factory:  chainFactory,
```

Comment:
this is not directly in the scope of this PR, but I would find it much easier to follow (especially since there are a lot of different factory thing) if `factory` would be renamed to `chainFactory`.